src/wp-includes/block-template-utils.php

```diff
@@ -224,14 +224,17 @@ static function ( $item ) {
  * @return string[] A list of paths to all template part files.
  */
 function _get_block_templates_paths( $base_directory ) {
+	static $wp_path_list = array();
+	if ( isset( $wp_path_list[ $base_directory ] ) ) {
+		return $wp_path_list[ $base_directory ];
+	}
 	$path_list = array();
-	if ( file_exists( $base_directory ) ) {
```

Comment:
Removed this check as `RecursiveDirectoryIterator` does file check almost immediately, giving no benefit as such on doing a separate file check   

Ref: 
RecursiveDirectoryIterator implementation:
https://github.com/php/php-src/blob/62e2402534d5e51faba41e30c79f8e4af1783370/ext/spl/spl_directory.c#L1475
File check line 
https://github.com/php/php-src/blob/62e2402534d5e51faba41e30c79f8e4af1783370/ext/spl/spl_directory.c#L1485

Other functions RecursiveIteratorIterator and RegexIterator also bails out early when the don't get proper data, making it more efficient not to have check at all.